tls/s2n_security_policies.c

```diff
@@ -784,3 +800,20 @@ int s2n_validate_kem_preferences(const struct s2n_kem_preferences *kem_preferenc
 
     return S2N_SUCCESS;
 }
+
+S2N_RESULT s2n_validate_certificate_signature_preferences(const struct s2n_signature_preferences *certificate_signature_preferences)
+{
+    ENSURE_REF(certificate_signature_preferences);
+
+    size_t rsa_pss_scheme_count = 0;
+
+    for (size_t i = 0; i < certificate_signature_preferences->count; i++) {
+        if (certificate_signature_preferences->signature_schemes[i]->libcrypto_nid == NID_rsassaPss) {
+            rsa_pss_scheme_count++;
+        }
+    }
+
+    /* Require all rsa pss signature schemes or none */
```

Comment:
Can we explain why in this comment? This is a strange check to be making, so we should document the reasoning.